conans/client/rest/file_downloader.py

```diff
@@ -86,24 +95,42 @@ def write_chunks(chunks, path):
                 ret = bytes(ret_data)
             return ret, downloaded_size
 
+        def get_total_length():
+            if range_start:
+                content_range = response.headers.get("Content-Range", "")
+                match = re.match(r"^bytes (\d+)-(\d+)/(\d+)", content_range)
+                if not match or range_start != int(match.group(1)):
+                    raise ConanException("Error in resumed download from %s\n"
+                                         "Incorrect Content-Range header %s" % (url, content_range))
+                return int(match.group(3))
+            else:
+                total_size = response.headers.get('Content-Length') or len(response.content)
+                return int(total_size)
+
         try:
             logger.debug("DOWNLOAD: %s" % url)
-            total_length = response.headers.get('content-length') or len(response.content)
-            total_length = int(total_length)
-            description = "Downloading {}".format(os.path.basename(file_path)) if file_path else None
+            total_length = get_total_length()
+            action = "Downloading" if range_start == 0 else "Continuing download of"
+            description = "{} {}".format(action, os.path.basename(file_path)) if file_path else None
             progress = progress_bar.Progress(total_length, self._output, description)
+            progress.initial_value(range_start)
 
             chunk_size = 1024 if not file_path else 1024 * 100
-            encoding = response.headers.get('content-encoding')
-            gzip = (encoding == "gzip")
-
             written_chunks, total_downloaded_size = write_chunks(
                 progress.update(read_response(chunk_size)),
                 file_path
             )
 
             response.close()
-            if total_downloaded_size != total_length and not gzip:
+            if (
+                file_path and total_length > total_downloaded_size > range_start
+                and response.headers.get("Accept-Ranges") == "bytes"
+            ):
+                written_chunks = self._download_file(url, auth, headers, file_path, try_resume=True)
+            elif (
+                total_downloaded_size != total_length
+                and response.headers.get("Content-Encoding") != "gzip"
```

Comment:
probably, better to check if ANY content encoding is enabled, not just gzip one

Reply:
also, consider checking `Transfer-Encoding` as well

Reply:
and also probably need to disable for `Multipart` content-type as well

Reply:
But those things are not related to this PR at all, they were already there, right?

Reply:
what do you mean? these are newly added lines

Reply:
Yeah, I condensed the logic from original lines 97, 98 and 106.

Reply:
Yes, that is what I meant. Those lines were there before this PR, just moved. If those things need to be changed/fixed, better be done in a separate PR. Please @SSE4 open an issue to log them for later work.